core/internal/pathtree/pathtree.go

```diff
@@ -4,13 +4,19 @@ import (
 	"fmt"
 
 	"github.com/segmentio/encoding/json"
+	// TODO: use simplejsonext for now until we replace the usage of json with
+	// protocol buffer and proto json marshaler
+	jsonext "github.com/wandb/simplejsonext"
 	"gopkg.in/yaml.v3"
 )
 
-// PathItem is a key-value pair with a path.
-type PathItem struct {
-	Path  []string
-	Value string
+// A type alias for any value that is passed from the client.
+//
+// It included HistoryItem, SummaryItem, ConfigItem, etc.
+type item interface {
```

Comment:
why is this being added back? any code that uses PathTree should use the appropriate struct instead of this

Reply:
Otherwise we duplicate the transformation, it can be factored out to history and summary just annoying